firefox-ios/Client/Frontend/TrackingProtection/TrackingProtectionViewController.swift

```diff
@@ -863,6 +885,32 @@ class TrackingProtectionViewController: UIViewController, Themeable, Notifiable,
 
     func willDismiss() {
     }
+
+    // MARK: - Update Views
+    private func updateProtectionViewStatus() {
+        switch toggleSwitch.isOn {
+        case true:
+            // TODO: FXIOS-9195 #20363 Enhanced Tracking Protection screen status enabled design
+            toggleStatusLabel.text = .Menu.EnhancedTrackingProtection.switchOnText
+            trackersView.isHidden = false
+            trackersLabelTopConstraint?.constant = TPMenuUX.UX.trackersLabelConstraintConstant
+            trackersLabelBottomConstraint?.constant = -TPMenuUX.UX.trackersLabelConstraintConstant
+            viewModel.isProtectionEnabled = true
+            foxStatusImage.image = viewModel.protectionOnImage
+            connectionDetailsContentView.backgroundColor = currentTheme().colors.layerAccentPrivateNonOpaque
+
+        case false:
+            toggleStatusLabel.text = .Menu.EnhancedTrackingProtection.switchOffText
+            trackersView.isHidden = true
+            trackersLabelTopConstraint?.constant = 0
+            trackersLabelBottomConstraint?.constant = 0
+            viewModel.isProtectionEnabled = false
+            foxStatusImage.image = viewModel.protectionOffImage
+            connectionDetailsContentView.backgroundColor = currentTheme().colors.layer3
+        }
```

Comment:
Here, the things are different. We may update the colour based on the Switch statement.

Reply:
I was thinking we can add that logic to `applyTheme` so that everything lives in a central location. It seems we have it in two places now. Do we still need to keep the one here?

Reply:
Yes, we still need to keep it. :)